poptorch_experimental_addons/_impl/collectives.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+import torch
+import poptorch
+
+
+def replicated_all_gather(x: torch.Tensor, replication_factor: int) -> torch.Tensor:
+    """
+    SPMD all-gather across IPUs.
```

Comment:
Again, for consistency - I'm happy to use this docstring style, but I'd like to be consistent in `sparse.py`.

Reply:
Yes - addressed in #4 to match this